text/0037-ternary-numeric-conversion.md

```diff
@@ -26,28 +26,81 @@ assert_eq!(x, y);
 This RFC introduces the following trait implementations:
 
 ```rust
-// Binary to ternary
-impl<I: AsPrimitive<i64>, T: RawEncodingBuf> From<I> for TritBuf<T> {}
+// Signed binary to balanced ternary
+impl<T: RawEncodingBuf> From<i64> for TritBuf<T> where T::Slice: RawEncoding<Trit = Btrit> {}
+impl<T: RawEncodingBuf> From<i32> for TritBuf<T> where T::Slice: RawEncoding<Trit = Btrit> {}
+impl<T: RawEncodingBuf> From<i16> for TritBuf<T> where T::Slice: RawEncoding<Trit = Btrit> {}
+impl<T: RawEncodingBuf> From<i8> for TritBuf<T> where T::Slice: RawEncoding<Trit = Btrit> {}
 
-// Ternary to binary (these cannot be implemented as above due to Rust's orphan rules)
+// Balanced ternary to signed binary
 impl<'a, T: RawEncoding<Trit = Btrit> + ?Sized> TryFrom<&'a Trits<T>> for i64 {}
 impl<'a, T: RawEncoding<Trit = Btrit> + ?Sized> TryFrom<&'a Trits<T>> for i32 {}
 impl<'a, T: RawEncoding<Trit = Btrit> + ?Sized> TryFrom<&'a Trits<T>> for i16 {}
 impl<'a, T: RawEncoding<Trit = Btrit> + ?Sized> TryFrom<&'a Trits<T>> for i8 {}
+
+// Unsigned binary to unbalanced ternary
+impl<T: RawEncodingBuf> From<u64> for TritBuf<T> where T::Slice: RawEncoding<Trit = Utrit> {}
+impl<T: RawEncodingBuf> From<u32> for TritBuf<T> where T::Slice: RawEncoding<Trit = Utrit> {}
+impl<T: RawEncodingBuf> From<u16> for TritBuf<T> where T::Slice: RawEncoding<Trit = Utrit> {}
+impl<T: RawEncodingBuf> From<u8> for TritBuf<T> where T::Slice: RawEncoding<Trit = Utrit> {}
+
+// Unbalanced ternary to unsigned binary
+impl<'a, T: RawEncoding<Trit = Utrit> + ?Sized> TryFrom<&'a Trits<T>> for u64 {}
+impl<'a, T: RawEncoding<Trit = Utrit> + ?Sized> TryFrom<&'a Trits<T>> for u32 {}
+impl<'a, T: RawEncoding<Trit = Utrit> + ?Sized> TryFrom<&'a Trits<T>> for u16 {}
+impl<'a, T: RawEncoding<Trit = Utrit> + ?Sized> TryFrom<&'a Trits<T>> for u8 {}
 ```
 
 The aforementioned trait implementations allow conversion to and from a variety
 of numeric binary types for ternary types. The `AsPrimitive` trait comes from
```

Comment:
Since the line 30 removes the usage of `AsPrimitive` the explanation where it comes from isn't what the reader expects here. I suggest moving it further down after the code where it actually appears.

Reply:
Good catch, changing